app/Providers/ApiClientServiceProvider.php

```
@@ -0,0 +1,139 @@
+            // Create and configure the base GitHub client
+            $factory = $app->make(GitHubFactory::class);
+            $client = $factory->make($config);
+
+            $authentication = $this->resolveAuthenticationPayload($config);
+            if ($authentication !== null) {
+                [$identifier, $secret, $method] = $authentication;
+                $client->authenticate($identifier, $secret, $method);
+            }
+
+            if ($config['cache'] ?? true) {
+                $client->addCache(Cache::pool());
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Support\Facades\Cache' in method 'register'. (undefined)

(StaticAccess)